test/conformance/device_code/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ macro(add_device_binary SOURCE_FILE)
             COMMAND ${UR_DPCXX} -fsycl -fsycl-targets=${TRIPLE} -fsycl-device-code-split=off
             ${SOURCE_FILE} -o ${EXE_PATH}
             COMMAND ${CMAKE_COMMAND} -E env SYCL_DUMP_IMAGES=true
-            ${EXE_PATH} || (exit 0)
```

Comment:
That's a good point... The build also fails right now because generating `kernel_entry_points.h` is part of the `device_code` cmake.

Approaching this from a different side, there are two 'build criteria' I see for kernel tests. First, as you've noted, `ld` needs to be able to find libsycl, and the second is having a device capable of running the kernels for the triplet.
If we can't rely on the binary failing, I think we need to detect those criteria prior to running it and emit a relevant message.